core/src/test/java/hudson/util/RobustReflectionConverterTest.java

```diff
@@ -84,6 +86,20 @@ public void ifWorkaroundNeeded() {
         assertThat(e.getMessage(), containsString("No such field hudson.util.Point.z"));
     }
 
+    private Map<String, String> getClassOwnerMap(String target) {
+        Map<String, String> classOwnerMap = new TreeMap<>();
+        int index = target.indexOf("plugin=", 0);
+        while (index != -1) {
+            int classNameStart = target.lastIndexOf("<", index) + 1;
+            String className = target.substring(classNameStart, index - 1);
+            int ownerNameEnd = target.indexOf("'", index + 8);
+            String ownerName = target.substring(index + 8, ownerNameEnd);
+            classOwnerMap.put(className, ownerName);
+            index = target.indexOf("plugin=", index + 1);
+        }
+        return classOwnerMap;
+    }
```

Comment:
Parsing the XML is complicated and fragile. Would be simpler to merely construct the `Map` first and then build the XML string based on that.

Reply:
Do you mean to build a `Map` that repersent the structure of the object, parse it to XML, and compare the produced XML string directly with the XML string generated by parsing the Object? I think it's kind of hard to repersent the `Ownership` part, which is the key of this testing, inside of `Map`.

However, I am not too familiar with the project. If I said something wrong, please point out.